mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -5245,6 +5245,30 @@ pub enum CancelBy {
     Coin { ticker: String },
 }
 
+pub async fn get_matching_orders(ctx: &MmArc, coins: &HashSet<String>) -> Result<Vec<Uuid>, String> {
+    let ordermatch_ctx = try_s!(OrdermatchContext::from_ctx(ctx));
+    let mut matching_orders = vec![];
+
+    let maker_orders = ordermatch_ctx.maker_orders_ctx.lock().orders.clone();
+    let taker_orders = ordermatch_ctx.my_taker_orders.lock().await;
+    for ticker in coins {
+        for (uuid, order) in maker_orders.iter() {
```

Comment:
In this case, the complexity of such iteration is `O(coins * maker_orders)` that is too high.

The main idea was to iterate over all `maker_orders` and check that either `order.base` or `order.rel` is in the `coins: &HashSet<String>` container. `HashSet` allows to get an access to an element for a constant time O(1).

So in the way I suggested it should be `O(maker_orders)`

Reply:
Ah sorry, I missed the point.. got it